src/generate.ts

```diff
@@ -45,18 +45,23 @@ export async function generate(
   const generatorOutputValue = generator.output?.value;
   ok(generatorOutputValue, 'Missing generator configuration: output');
 
+  const config = createConfig(generator.config);
+
   const eventEmitter = new AwaitEventEmitter();
   eventEmitter.on('Warning', warning);
-  eventEmitter.on('Model', modelData);
-  eventEmitter.on('EnumType', registerEnum);
-  eventEmitter.on('OutputType', outputType);
-  eventEmitter.on('ModelOutputType', modelOutputType);
-  eventEmitter.on('AggregateOutput', createAggregateInput);
-  eventEmitter.on('InputType', inputType);
-  eventEmitter.on('ArgsType', argsType);
+  config.emitBlocks.models && eventEmitter.on('Model', modelData);
+  if (config.emitBlocks.prismaEnums || config.emitBlocks.schemaEnums) {
+    eventEmitter.on('EnumType', registerEnum);
+  }
```

Comment:
There are two types of enum classes generated - schema enums, the ones that are defined in schema.prisma, and prisma enums - enums used for other classes, like order by or fields etc. Prisma enums are used only in input and args classes, so it doesn't make sense to generate them if we want only models (which depend on schema enums)